### DISCUSSION TITLE
does file order in pre-processing code matter?

### DISCUSSION THREAD

**Discussion by samaramiller** [miniscope/minian#189] [Q&A]
Hello, I'm trying out the Minian pipeline and it look likes the files are being loaded out of order in the pre-processing step (I included snippet of the output code below). Is this a literal readout of how the files will be concatenated and analyzed throughout the rest of the pipeline. If so, how can I ensure the video files are loaded, sorted and concatenated in the correct order? Thank you!

`Input #0, avi, from '/Users/samaramiller/demo_movies/msCam4.avi':
  Metadata:
    encoder         : Lavf58.12.100
  Duration: 00:00:10.00, start: 0.000000, bitrate: 57762 kb/sInput #0, avi, from '/Users/samaramiller/demo_movies/msCam2.avi':
  Metadata:
    encoder         : Lavf58.12.100
  Duration: 00:00:10.00Input #0, avi, from '/Users/samaramiller/demo_movies/msCam3.avi':
, start: 0.000000  Metadata:
, bitrate:     encoder         : 
Lavf58.12.100
  Duration: 00:00:10.0057762 kb/s, start: 
0.000000, bitrate: 57762 kb/s
Input #0, avi, from '/Users/samaramiller/demo_movies/msCam1.avi':
  Metadata:
    encoder         : Lavf58.12.100
  Duration: 00:00:10.00, start: 0.000000, bitrate: 57762 kb/s`

**Comment by phildong** [comment]
hey the output you see is from ffmpeg, which is executed in parallel and it's totally expected to be asynchronous/out-of-order.
Minian use [natsort](https://pypi.org/project/natsort/) to determine the order of all the files that satisfy the filename patterns.
So as long as your filenames are in natural order it should be fine.
I would also recommend checking all the visualizations to make sure things are in order.
Let me know if you have more questions!
